lib/puppet/type/http.rb

```diff
@@ -1,17 +1,53 @@
-Puppet::Type.newtype :http, :is_capability => true do
+Puppet::Type.newtype :http do
+  @doc = "A type for testing http connections"
+
+  ensurable do
+    defaultvalues
+    defaultto :present
+  end
+
   newparam(:name, :is_namevar => true) do
-    desc 'The name of the resource'
+    desc 'The unique name of the resource'
+  end
+
+  newparam(:uri) do
+    desc 'The URL to test'
+    defaultto "http://127.0.0.1/"
   end
 
-  newparam(:ip) do
-    desc 'IP address of the HTTP service'
+  newparam(:timeout) do
+    desc 'Time before timing out the resource (seconds)'
+    defaultto 60
+
+    validate do |value|
+      Float(value)
+    end
+
+    munge do |value|
+      Float(value)
+    end
   end
 
-  newparam(:port) do
-    desc 'Port of the HTTP service'
+  newparam(:ping_interval) do
+    desc 'Time to sleep before attempting to try a connection again (second)'
+    defaultto 1
+
+    validate do |value|
+      Float(value)
+    end
+
+    munge do |value|
+      Float(value)
+    end
   end
 
-  newparam(:host) do
-    desc 'Hostname of the HTTP service'
+  newparam(:status_codes) do
+    desc 'An array of HTTP status codes that will return success. For example
+          [200, 302] will match for OK and redirection responses.'
+    defaultto [200]
```

Comment:
So seperately, so you want some way to catch 'any status' ...? I can dig that.

As far as making that default ... I would almost agree with you, if I didn't think that a 200 would presume a fully working system by default, whereas any other number would imply not-start/not-ready by default and is perhaps (while easier sometimes) more accurate/real-world-ready as a default.

But perhaps you are right. Following the familiar door-knob principle, perhaps I'll research what is the default through other testing systems too. In the case of the F5 system, they don't match status_codes, they match return strings instead: https://support.f5.com/kb/en-us/products/big-ip_ltm/manuals/product/ltm_configuration_guide_10_0_0/ltm_appendixa_monitor_types.html (see receive_string).